### DISCUSSION TITLE
BrowserFunction doesn't seem to be registered intermittently from Windows11/Edge/Webview2 on v4.31

### DISCUSSION THREAD

**Discussion by scottkurz** [eclipse-platform/eclipse.platform.swt#1274] [Q&A]
Running v4.31 on Windows 11 with Edge/Webview2, I have code that schematically might simplify to something like:

```java
Browser browser = new Browser(parent, SWT.EDGE);
String html = ... ;   // load HTML with React, etc.
browser.setText(html);

new MyBrowserFunction (browser, "myFunction");  // extends BrowserFunction
```

Once in awhile, the app renders in the browser but it seems like the BrowserFunction is simply not registered.   I don't think I've proven this really, from the webview2 browser process side, but that's what it seems like.

I was wondering if maybe the problem might be that you should really only register the BrowserFunction after say registering a ProgressListener and getting a `completed()` call.   But then I didn't find any doc saying that, though I did find it as maybe a workaround in this issue, which seemed to have an overlap with my problem.

https://bugs.eclipse.org/bugs/show_bug.cgi?id=578016#c5

Does this sound like it could be the problem?  Any other ideas on how to debug an intermittently non-functional BrowserFunction?  

Thank you.

**Comment by wimjongman** [comment]
Please provide a short selfcontained snippet that demonstrates the problem.

**Comment by scottkurz** [comment]
I haven't had time to reproduce with a snippet... but, in case it helps anyone in the meantime, let me just note that this idea I mentioned of registering a ProgressListener and then constructing the BrowserFunction in the method `org.eclipse.swt.browser.ProgressListener.completed(ProgressEvent)`  seems to consistently eliminate the problem.   

I'll also note that the problem occurred specifically when I had two separate views active at the same time, each using an SWT.EDGE Browser instance.   With the same code, it always worked fine as long as only one Browser-based view was active at a given time.

If it weren't for the fact that it works with a single view, I'd say it looks like I was creating the BrowserFunction too early.  But in truth, I haven't really proven much, but again, still thought this could be helpful to someone searching.
